docs/source/tutorial/05-connect-queries-to-your-ui.mdx

```diff
@@ -0,0 +1,61 @@
+---
+title: "5. Connect your queries to your UI"
+---
+
+In this chapter, you are going to display a list of Launch Sites in a [RecyclerView](https://developer.android.com/guide/topics/ui/layout/recyclerview).
+
+## Configure LaunchListAdapter
+
+Open `LaunchListAdapter` and add a `launches` property:
+
+```kotlin:title=app/src/main/java/com/example/rocketreserver/LaunchListAdapter.kt
+class LaunchListAdapter(val launches: List<LaunchListQuery.Launch>) :
+    RecyclerView.Adapter<LaunchListAdapter.ViewHolder>() {
+```
+
+`LaunchListQuery.Launch` is a typesafe generated model from your LaunchList.graphql query.
+
+## Bind the data to the UI
+
+Use the `launches` property to bind items to your adapter:
+
+```kotlin:title=app/src/main/java/com/example/rocketreserver/LaunchListAdapter.kt
+    override fun getItemCount(): Int {
+        return launches.size
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        val launch = launches.get(position)
+        holder.binding.site.text = launch.site ?: ""
+    }
+```
+
+## Pass the LaunchListAdapter to the RecyclerView
+
+Create a new Adapter and pass it to your recycler view:
+
+```kotlin:title=app/src/main/java/com/example/rocketreserver/LaunchListFragment.kt
+        lifecycleScope.launchWhenResumed {
+            val response = try {
+                apolloClient.query(LaunchListQuery()).toDeferred().await()
+            } catch (e: ApolloException) {
+                Log.d("LaunchList", "Failure", e)
+                null
+            }
+
+            val launches = response?.data?.launches?.launches?.filterNotNull()
+            if (launches != null && !response.hasErrors()) {
```

Comment:
should we cover the errors? To explain that user should check for response.errors?

Reply:
That'd be great, I'll see if I can trigger some meaningful error

Reply:
I added error handling to the details view, it's easy to trigger errors there by passing wrong ids: 

See https://jolly-kowalevski-0304f0.netlify.app/docs/android/tutorial/08-add-a-details-view/#handle-protocol-errors for a preview